src/context.rs

```diff
@@ -887,6 +892,27 @@ impl Context {
         Ok(mvbox.as_deref() == Some(folder_name))
     }
 
+    pub(crate) async fn should_delete_to_trash(&self) -> Result<bool> {
+        if let Some(v) = self.get_config_parsed::<i8>(Config::DeleteToTrash).await? {
+            return Ok(v != 0);
+        }
+        if let Some(provider) = self.get_configured_provider().await? {
+            return Ok(provider.opt.delete_to_trash);
+        }
+        Ok(false)
+    }
+
+    /// Returns `target` for deleted messages as per `imap` table. Empty string means "delete w/o
+    /// moving to trash".
+    pub(crate) async fn get_delete_msgs_target(&self) -> Result<String> {
+        if !self.should_delete_to_trash().await? {
+            return Ok("".into());
+        }
+        self.get_config(Config::ConfiguredTrashFolder)
```

Comment:
Yes, but i added the code updating `Config::ConfiguredTrashFolder` value to `Imap::scan_folders()` where the same code for `Config::ConfiguredSentboxFolder` is. Looks working for my Gmail account

Reply:
But yes, until `scan_folders()` is done, there may be errors because of absent `ConfiguredTrashFolder` config parameter. So, increased `DC_FOLDERS_CONFIGURED_VERSION` also